datasetinsights/data/datasets/groceries_real.py

```diff
@@ -157,6 +176,86 @@ def _filepath(self, filename):
         """
         return os.path.join(self.root, self.version, filename)
 
+    @staticmethod
+    def _download_http(source_uri, dest_path, version):
+        """ Download dataset from Public HTTP URL.
+
+        Args:
+            source_uri (str): source url where the file should be downloaded
+            dest_path (str): destination path of the file
+
+        Raises:
+            DownloadError if a given definition path can't be found
+            ChecksumError if the downloaded file checksum does not match
+        """
+
+        try:
+            logger.info("Downloading the dataset.")
+            download_file(source_uri=source_uri, dest_path=dest_path)
+        except DownloadError as e:
+            logger.info(
+                f"The request download from {source_uri} -> {dest_path} can't "
+                f"be completed."
+            )
+            raise e
+        expected_checksum = GroceriesReal.GROCERIES_REAL_DATASET_TABLES[
+            version
+        ].checksum
+        try:
+            validate_checksum(dest_path, expected_checksum)
+        except ChecksumError as e:
+            logger.info("Checksum mismatch. Delete the downloaded files.")
+            os.remove(dest_path)
+            raise e
+
+    @staticmethod
+    def _extract_file(dest_path, root_dir):
+        """ Unzip the downloaded file.
+        """
+        logger.info("Unzipping the dataset file.")
+        with zipfile.ZipFile(dest_path, "r") as zip_dir:
+            zip_dir.extractall(root_dir)
+
+    @staticmethod
+    def download(data_root, version):
+        """ Download dataset from Public HTTP URL.
+
+        If the file already exists and the checksum matches, it will skip the
+        download step. If not, it would delete the previous file and download
+        it again. If the file doesn't exist, it would download the file.
+        """
```

Comment:
This docstring is incomplete. What's input args and what's the retrun type? What's the expected error that might be raised?